### ISSUE TITLE
Syncing algorithm changes

### ISSUE THREAD

**Issue by orthecreedence** [turtl/tracker#317]
Right now, if I add a note on desktop, edit the note in mobile (offline mode), then edit the note again on desktop, the current behavior is that when the mobile device comes online, it will overwrite the latest changes the desktop saved to the note.

This is not desirable. Sync records should have some concept of ordering. If we're going to do "last write wins" then it should at least favor the last *temporal* write (and assume your clocks are set correctly).

In other words, if we see a note's last mod date is *after* the sync item was created, we should throw out that sync record. Same with files: if the note's file was updated after the create date of the new file, toss it out.
